### ISSUE TITLE
hypot function typescript definition seems wrong

### ISSUE THREAD

**Issue by silentmissile** [josdejong/mathjs#3125]
in typescript definition of mathjs, there is hypot function definition

hypot<T extends (number | BigNumber)[]>(...args: T[]): T

in this definition T can't be number, T has to be number[], and this function would return number[], this result is not our expectation

so I think it actually means:

hypot<T extends (number | BigNumber)>(...args: T[]): T

**Comment by josdejong** [comment]
Thanks for bringing this up.

Function `hypot` can be both:

```
math.hypot(a, b, ...)
math.hypot([a, b, c, ...])
```

See code: https://github.com/josdejong/mathjs/blob/develop/src/function/arithmetic/hypot.js#L25-L28

We should reflect that in the type definitions. Anyone able to help improve the definitions?

**Comment by Hudsxn** [comment]
The problem here is that if you were to specify number|BigInteger alongside its array type declaration, you'd incur an error explaining all the iterative methods we're missing from said new type. However, you can use function overloading for this. This is the process of having one implementation with multiple method signatures.

``` typescript
math.hypot<T extends (number|BigNumber)>(...args:T[]) : T;
math.hypot<T2 extends (number|BigNumber)[]>(...args:T2) : T;
```

In this example above, you're returning the same type, just altering generic parameters to accommodate what you can do in javascript naturally as its not type safe.

**Comment by josdejong** [comment]
Fixed now in `v12.3.1`
